### DISCUSSION TITLE
Translations not loading in on production with getServerSideProps and next-i18next

### DISCUSSION THREAD

**Discussion by thibaultferaux** [i18next/next-i18next#2159] [Q&A]
In Next.js I am using the package `next-i18next` for my translations.

All my translations are working fine and when I have a page with getServerSideProps it works on my local development server but when I push it to production with Vercel the translations on those pages are not working. On pages where I use getStaticProps it works.

FYI: I use the pages router in Next.js.

This is my getServerSideProps function:

```javascript
export async function getServerSideProps({ query, locale }) {
  const directus = await getDirectusClient()
  const { id } = query;
  const quote = await directus.items("Quotes").readOne(id, {
    fields: ["*.*"],
  });

  if (!quote) {
    return {
      notFound: true,
    };
  }

  return {
    props: {
      quote,
      ...(await serverSideTranslations(locale, ["common", "nav", "portal", "footer", "configurations"])),
    },
  };
}
```

This is how I call the translations in the component

```javascript
const { t } = useTranslation(["portal", "configurations", "common"]);
```

I render a translation by: `<p>{ t('quote details') }</p>`

Is there anyone who recognizes this problem and knows how to solve it?

I tried it on another component that was working with getStaticProps, and changed it to getServerSideProps. I wanted to know if it was a problem with the component or not but on the component that was working before it didn't work and just rendered the name of the labels from the translations :(.

**Comment by adrai** [comment]
This seems to work: https://github.com/i18next/next-i18next/blob/9b44b39abece1533423de6582c7c2961dbaa3b33/examples/simple/pages/second-page.tsx#L40
deployed to vercel: https://next.i18next.com/second-page
